packages/editor/src/components/post-actions/actions.js

```diff
@@ -1073,8 +1076,10 @@ export function usePostActions( { postType, onActionPerformed, context } ) {
 				  ! isPattern &&
 				  duplicatePostAction
 				: false,
-			isTemplateOrTemplatePart && duplicateTemplatePartAction,
-			isPattern && duplicatePatternAction,
+			isTemplateOrTemplatePart &&
+				userCanCreatePostType &&
+				duplicateTemplatePartAction,
+			isPattern && userCanCreatePostType && duplicatePatternAction,
```

Comment:
Are these both saved as `blocks` CPTs?

Reply:
With that I mean: both patterns and template parts?

Reply:
No, patterns are on wp_block/blocks and template parts are on wp_template_part/template_parts. But we are checking if the user can create the post type the actions are being used on userCanCreatePostType: canUser( 'create', resource ). So the same condition can apply to both.